src/coreclr/vm/method.cpp

```diff
@@ -3258,7 +3095,14 @@ void MethodDesc::SetCodeEntryPoint(PCODE entryPoint)
     }
     else if (!HasStableEntryPoint())
     {
-        SetStableEntryPointInterlocked(entryPoint);
+        if (RequiresStableEntryPoint())
+        {
+            GetOrCreatePrecode()->SetTargetInterlocked(entryPoint);
+        }
+        else
+        {
+            SetStableEntryPointInterlocked(entryPoint);
+        }
```

Comment:
This code reads odd. If `RequiresStableEntryPoint()` returns `false`, then we set the stable entry point? If this is correct, I think a comment is appropriate.

Reply:
Sigh. It is right, but yeah, this should be better structured. A RequiredStableEntryPoint is currently required have a stable entrypoint which is a Precode. Otherwise, we can generate set a stable entrypoint which isn't based on a Precode. Before this change, we would have reliably hit the HasPrecode flag above, but with making Precode generation lazy, this is a point where the Precode needs to be forced at this point. However, I think I'll fix this by changing the previous condition to be (RequiresStableEntryPoint()) and not use GetPrecode, and instead use GetOrCreatePrecode(). Then this portion of the condition can revert to be what it was before.